core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/date/DateRangeAggregationBuilder.java

```diff
@@ -285,7 +281,25 @@ protected DateRangeAggregatorFactory innerBuild(SearchContext context, ValuesSou
             AggregatorFactory<?> parent, Builder subFactoriesBuilder) throws IOException {
         // We need to call processRanges here so they are parsed and we know whether `now` has been used before we make
         // the decision of whether to cache the request
-        Range[] ranges = processRanges(context, config);
+        RangeAggregator.Range[] ranges = processRanges(range -> {
+            DocValueFormat parser = config.format();
+            assert parser != null;
+            double from = range.getFrom();
+            double to = range.getTo();
+            String fromAsString = range.getFromAsString();
+            String toAsString = range.getToAsString();
+            if (fromAsString != null) {
+                from = parser.parseDouble(fromAsString, false, context.getQueryShardContext()::nowInMillis);
+            } else if (Double.isFinite(from)) {
+                from = parser.parseDouble(Long.toString((long) from), false, context.getQueryShardContext()::nowInMillis);
```

Comment:
In that case, if we are always going to interpret the users input as a formatted date, then we should just have the parser get the string value of the from and to fields in the request and not have the parser try to work out if the input is a string or a number and convert to string at that point. That will mean we don't have the conversion `double --> long --> string --> double` and instead always have `string --> double`. 

Saying that I'm not sure if the original intended fix to #17920 was to also remove the ability to specific epoch time as a long when the format is non-epoch based /cc @clintongormley